optional-clustering-analysis/clustering-calculations.R

```diff
@@ -0,0 +1,208 @@
+    help = "Method used for clustering. Can be either louvain or walktrap.",
+  ),
+  optparse::make_option(
+    c("-n", "--nearest_neighbors_range"),
+    type = "integer",
+    default = 5:25,
+    help = "Range with number of nearest neighbors to include during graph 
+    construction. Can be a range of values, e.g. 5:25, or a single value."
+  ),
+  optparse::make_option(
+    c("-i", "--nearest_neighbors_increment"),
+    type = "integer",
+    default = 5,
+    help = "Increment to use when implementing the range number of nearest 
+    neighbors for calculating the cluster stats. If using a single value for the
+    nearest neighbors range, set to NULL"
+  ),
+  optparse::make_option(
+    c("-o", "--output_directory"),
+    type = "character",
+    default = NULL,
+    help = "path to output directory that would hold the RDS file containing SCE
+    with cluster assignments added, as well as the TSV files with the clustering stats"
+  ),
+  optparse::make_option(
+    c("--project_root"),
+    type = "character",
+    default = NULL,
+    help = "the path to the root directory for the R project and where the `utils` folder lives."
+  )
+)
+
+# Read the arguments passed
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# if project root is not provided use here::here()
+if(is.null(opt$project_root)){
+  project_root <- here::here()
+} else {
+  project_root <- opt$project_root
+}
+
+# Source in set up function
+source(file.path(project_root, "utils", "setup-functions.R"))
+
+# source in clustering functions 
+source(file.path(project_root, "utils", "clustering-functions.R"))
+
+# Check R and Bioconductor versions
+check_r_bioc_versions()
+
+# Load project
+setup_renv(project_filepath = project_root)
+
+## Load libraries
+suppressPackageStartupMessages({
+  library(magrittr)
+  library(bluster)
+  library(SingleCellExperiment)
+})
+
+## Set the seed
+set.seed(opt$seed)
+
+# Check that the input file exists
+if (!file.exists(opt$sce)){
+  stop(paste(opt$sce, "does not exist."))
+}
+
+# Check that clustering type is valid 
+if (!opt$cluster_type %in% c("louvain", "walktrap")) {
+  stop("--cluster_type (-c) must be either louvain or walktrap.")
+}
+
+# Make sure the output directory exists 
+if(!dir.exists(opt$output_directory)){
+  dir.create(opt$output_directory, recursive = TRUE)
+}
+
+stats_output_dir <- file.path(opt$output_directory, "clustering_stats")
+if(!dir.exists(stats_output_dir)){
+  dir.create(stats_output_dir, recursive = TRUE)
+}
+
+#### Read in data and check formatting -----------------------------------------
+
+# Read in normalized sce object
+sce <- readr::read_rds(opt$sce)
+
+# Check that the input data contains an SCE with PCA results 
+if(is(sce,"SingleCellExperiment")){
+  if(!"PCA" %in% reducedDimNames(sce)) {
+    stop("PCA results are not found in the provided SingleCellExperiment object.")
+  }
+} else {
+  stop("file specified by --sce (-s) must contain a SingleCellExperiment object.")
```

Comment:
This conflicts. You used `-i` for the short opt, but `-s` for the short seed opt. 
```suggestion
  stop("file specified by --sce (-i) must contain a SingleCellExperiment object.")
```